src/splat/segtypes/segment.py

```diff
@@ -108,21 +108,35 @@ def get_class_for_type(seg_type) -> Type["Segment"]:
     @staticmethod
     def get_base_segment_class(seg_type):
         platform = options.opts.platform
+        ext_path = options.opts.extensions_path
         is_platform_seg = False
 
-        # heirarchy is platform -> common -> fail
+        # heirarchy is unsupported platform -> platform -> common -> fail
         try:
-            segmodule = importlib.import_module(
-                f".segtypes.{platform}.{seg_type}", package=__package_name__
+            assert options.opts.unsupported_platform
+            assert ext_path is not None
+            ext_spec = importlib.util.spec_from_file_location(
+                f"{__package_name__}.segtypes.{platform}.{seg_type}",
+                ext_path / platform / f"{seg_type}.py",
             )
+            assert ext_spec is not None
+            segmodule = importlib.util.module_from_spec(ext_spec)
+            assert ext_spec.loader is not None
+            ext_spec.loader.exec_module(segmodule)
             is_platform_seg = True
-        except ModuleNotFoundError:
+        except Exception:
```

Comment:
Why is this new block necessary? Shouldn't the old code just look up for the extensions instead?

Reply:
This is for the unsupported platform defined segments in the extension folder to take precedence over the splat builtin platform segments

Reply:
But extensions take precedence already, don't they?

Reply:
nope